### PR TITLE
Add Support for non default keystore and truststore type in presto CLI and JDBC

### PR DESCRIPTION
**Pull request by evanvdia**
## Description

Added Support for non default keystore and truststore type in presto CLI and JDBC.

## Motivation and Context

Issue : https://github.com/prestodb/presto/issues/22555

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

General Changes
    * Add support for non default keystore and truststore type.
```

### CODE REVIEW

**Comment by steveburnett** [comment]
Suggest changing title, Description, release note entry, and commit message from "keystroke" to "keystore", based on the edited files and on the Desription in [Issue 22555](https://github.com/prestodb/presto/issues/22555).

**Comment by evanvdia** [comment]
@steveburnett changes done

File: presto-client/src/main/java/com/facebook/presto/client/OkHttpUtil.java [hunk 1]
```diff
@@ -178,7 +180,7 @@ public static void setupSsl(
             // load TrustStore if configured, otherwise use KeyStore
             KeyStore trustStore = keyStore;
             if (trustStorePath.isPresent()) {
-                trustStore = loadTrustStore(new File(trustStorePath.get()), trustStorePassword);
+                trustStore = loadTrustStore(new File(trustStorePath.get()), trustStorePassword, trustStoreType.get());
```
Comment (tdcmeehan) [inline presto-client/src/main/java/com/facebook/presto/client/OkHttpUtil.java:183]: Please add a validation to ensure that these parameters are present before getting.

**Review by tdcmeehan — CHANGES_REQUESTED** [review]

File: presto-client/src/main/java/com/facebook/presto/client/OkHttpUtil.java (hunk 1 above)
Comment (evanvdia) [reply presto-client/src/main/java/com/facebook/presto/client/OkHttpUtil.java:183]: We have set default values in ClientOptions.java

image


image

Comment (tdcmeehan) [reply presto-client/src/main/java/com/facebook/presto/client/OkHttpUtil.java:183]: Understood, but that's in a different module.  Best practice is to either fail, or only pass in String (and check not null).

**Comment by tdcmeehan** [comment]
Looks good, but the commit message is too long (see [contributing guide](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#commit-standards)).  Perhaps just mention `Add Support for non default keystroke and truststore type`

**Comment by evanvdia** [comment]
Changed commit message.

**Comment by evanvdia** [comment]
@tdcmeehan Review comments are incorporated. Could you please review.

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**